packages/salesforcedx-vscode-core/src/util/authInfo.ts

```diff
@@ -104,6 +104,25 @@ export class OrgAuthInfo {
     );
   }
 
+  public static async isASandboxOrg(username: string): Promise<boolean> {
+    const authInfo = await AuthInfo.create({ username });
+
+    const authInfoFields = authInfo.getFields();
+    return authInfoFields.instanceUrl?.includes('sandbox') || false;
```

Comment:
This is the best way I found to identify a sandbox. I have tried other solutions such as [AuthInfo.isSandBox](https://forcedotcom.github.io/sfdx-core/types/org_authInfo.AuthFields.html#__type.isSandbox) and [Org.isSandbox()](https://forcedotcom.github.io/sfdx-core/classes/org_org.Org-1.html#isSandbox) but they did not work as expected. I found the sandbox instanceUrl always works for me.

Reply:
@mingxuanzhangsfdx what do you mean by did not work as expected? Searching URLs for sandbox is not guaranteed over time.

Reply:
the isSandbox returns false when I literally uses a sandbox org.

Reply:
The sandbox property is a bit complicated. These CLI commands
```
org create sandbox                │ Create a sandbox org.                                                                                                                                                                                
org delete sandbox               │ Delete a sandbox.                                                                                                                                                                                    
org refresh sandbox              │ Refresh a sandbox org using the sandbox name.                                                                                                                                                        
org resume sandbox     
```
and when managing sandboxes the property is only set when these commands are used.

Another option that might be used in combination with isSandbox is to query the [Organization](https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_organization.htm) in the org in question.

It has a field called isSandbox, which is set during sandbox creation.